src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlTransactionScope.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Data.SqlClient;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
+{
+    internal class SqlTransactionScope : ITransactionScope
+    {
+        private readonly SqlTransactionHandler _sqlTransactionHandler;
+        private bool _isDisposed;
+
+        public SqlTransactionScope(SqlServerDataStoreConfiguration configuration, SqlTransactionHandler sqlTransactionHandler)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(sqlTransactionHandler, nameof(sqlTransactionHandler));
+
+            SqlConnection = new SqlConnection(configuration.ConnectionString);
+            SqlConnection.Open();
+
+            SqlTransaction = SqlConnection.BeginTransaction();
+
+            _sqlTransactionHandler = sqlTransactionHandler;
+        }
+
+        public SqlConnection SqlConnection { get; private set; }
+
+        public SqlTransaction SqlTransaction { get; private set; }
+
+        public void Complete()
+        {
+            SqlTransaction.Commit();
+        }
+
+        public void Dispose()
```

Comment:
should either mark the class as sealed or provide overridable Dispose (I thought FxCop would create warning for this?)

Reply:
It seems like `ca1063` should have fired for this, it's only ignored in test projects. I'll see if I can figure out whats suppressing it.

Reply:
It looks like we're missing a reference to `Microsoft.CodeAnalysis.FxCopAnalyzers` in a number of projects. I opened #756 to track it.

Reply:
Also, it appears that the rule doesn't fire when it's an internal class.